### ISSUE TITLE
Cannot read property 'body' of undefined.

### ISSUE THREAD

**Issue by aymather** [thelinmichael/spotify-web-api-node#354]
I am getting this application crash error which is very strange...? Are we just not getting a response sometimes?

```
/node_modules/spotify-web-api-node/src/http-manager.js:34
  if (typeof response.body === 'object' && response.body.error && typeof response.body.error === 'object' && response.body.error.reason) {
                      ^

TypeError: Cannot read property 'body' of undefined
    at _toError (/node_modules/spotify-web-api-node/src/http-manager.js:34:23)
    at /node_modules/spotify-web-api-node/src/http-manager.js:71:25
    at Request.callback (/node_modules/superagent/lib/node/index.js:905:3)
    at ClientRequest.<anonymous> (/node_modules/superagent/lib/node/index.js:822:12)
    at ClientRequest.emit (events.js:315:20)
    at TLSSocket.socketErrorListener (_http_client.js:469:9)
    at TLSSocket.emit (events.js:315:20)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by JonBons** [comment]
I'm seeing the same issue in a similar app. I've modified my copy of this module to log the `err ` object being used to pass the request data into the `_toError` function and this is the exception being thrown which appears to not include a response object:

```
Error: Client network socket disconnected before secure TLS connection was established
    at connResetException (internal/errors.js:609:14)
    at TLSSocket.onConnectEnd (_tls_wrap.js:1549:19)
    at Object.onceWrapper (events.js:420:28)
    at TLSSocket.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  code: 'ECONNRESET',
  path: null,
  host: 'api.spotify.com',
  port: 443,
  localAddress: undefined,
  response: undefined
}
```

**Comment by aymather** [comment]
@JonBons me as well, hopefully gets merged soon!

**Comment by thelinmichael** [comment]
Fixed in 5.0.1. Closing this. Thanks.

**Comment by voxsoftware** [comment]
Thanks. I were waiting more than one month for this solution
